### PR TITLE
[#26] unit tests with both elements and measure

### LINKED ISSUE #26: More robust unit testing with both elements and measure
Currently unit tests compare two treaps only by comparing elements via `toList` function. It would be more robust to test also the measure of the `Treap`. But for that we need more convenient util functions.

### PR DESCRIPTION
**Pull request by sphaso**
Resolves #26 

I might be able to avoid the parentheses by tweaking the infix priority. Let me know what you think :)

### CODE REVIEW

File: test/Test/Common.hs [hunk 1]
```diff
@@ -19,3 +22,14 @@ smallTreap = fromList [1..5]
 
 extractSum :: TestTreap -> Int
 extractSum = getSum . measure
+
+with :: [Int] -> Int -> ([Int], Sum Int)
+with l m = (l, Sum m)
+
+describedAs :: TestTreap -> ([Int], Sum Int) -> Expectation
+describedAs t nodesMeasure = (treapNodes, Sum treapMeasure) `shouldBe` nodesMeasure
```
Comment (chshersh) [inline test/Test/Common.hs]: Just minor suggestion for the function.
```suggestion
describedAs t expectedNodesMeasure
    = treapNodes `with` treapMeasure `shouldBe` expectedNodesMeasure
```

File: test/Test/Rand/Cuts.hs [hunk 2]
```diff
@@ -27,27 +26,30 @@ splitAtSpec = describe "splitAt" $ do
         snd (Treap.splitAt 0 smallTreap) `shouldBe` smallTreap
     it "splitAt n returns treap itself" $
         fst (Treap.splitAt 5 smallTreap) `shouldBe` smallTreap
-    it "splitAt 2 returns two treaps" $
-        bimap toList toList (Treap.splitAt 2 smallTreap) `shouldBe` ([1..2], [3..5])
+    it "splitAt 2 returns two treaps" $ do
+        let
+            (a, b) = Treap.splitAt 2 smallTreap
+        a `describedAs` ([1..2] `with` 3)
```
Comment (chshersh) [inline test/Test/Rand/Cuts.hs]: Yeah, assigning proper fixity priorities to the `with` function will make code nicer 🙂

File: test/Test/Rand/Update.hs [hunk 3]
```diff
@@ -33,6 +32,6 @@ deleteSpec = describe "delete" $ do
     it "delete size does nothing" $
         Treap.delete 5 smallTreap `shouldBe` smallTreap
     it "delete 0 removes first element" $
-        toList (Treap.delete 0 smallTreap) `shouldBe` [2..5]
+        Treap.delete 0 smallTreap `describedAs` ([2..5] `with` 14)
```
Comment (chshersh) [inline test/Test/Rand/Update.hs]: Thanks for careful update of all relevant tests!

**Review by chshersh — COMMENTED** [review]
Nice addition! I only think that the code can be slightly neater without `()` but overall looks great 👍

**Comment by sphaso** [comment]
@chshersh fixed the infix priority :)

**Review by chshersh — APPROVED** [review]
Nice! There are some conflicts now. But should be straightforward to resolve them. Afther that: ready to merge 🙂

**Comment by sphaso** [comment]
@chshersh conflicts are solved :) thanks for the review!

**Comment by chshersh** [comment]
@sphaso Thanks a lot for your work! :blush:

**Pull request merged**